### ISSUE TITLE
Slow response issue when opening project file

### ISSUE THREAD

**Issue by lyh061619** [apache/netbeans#6800]
### Apache NetBeans version

Apache NetBeans 20 release candidate

### What happened

Clicking on the menu file ->Open Project takes about 6-8 seconds for the dialog box to pop up. Secondly, clicking on the ski bar in the project tree to browse the project file experience is not very good, resulting in slow response issues,
![n1](https://github.com/apache/netbeans/assets/15118571/ba5383eb-a6a8-4fd2-9bbf-0b56500d0994)

Selfsampler File:
[selfsampler.zip](https://github.com/apache/netbeans/files/13601430/selfsampler.zip)

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

win10

### JDK

zulu21.28.85-ca-jdk21.0.0

### Apache NetBeans packaging

Apache NetBeans binary zip

### Are you willing to submit a pull request?

No

**Comment by lyh061619** [comment]
> This issue can't be acted upon:
> 
>     * the message log is missing and no error is reported
> 
>     * for "it is slow" problems, the IDE has a built-in profiler (Enable "Performance" symbol bar, first icon: ![image](https://private-user-images.githubusercontent.com/2179736/288492927-f078172e-ce58-4421-a4f7-17525aa437a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NjA2NzUsIm5iZiI6MTcwMTk2MDM3NSwicGF0aCI6Ii8yMTc5NzM2LzI4ODQ5MjkyNy1mMDc4MTcyZS1jZTU4LTQ0MjEtYTRmNy0xNzUyNWFhNDM3YTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDdUMTQ0NjE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWZjZjliNDZkNzNmZWVjNzk3MWVhZDg1NWRmNzNlMDI0YzBmMzY0YWRkNDc0YjdmN2M3MWM3NzA3MGRjMWUzNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.CFTlmZpSj75D6BhPeRnM1wkgHMVW1CWD83j5Ij94JmQ))
> 
>     * what is a "ski bar"?

I have added the issue log. This bug is really a fatal problem. It takes 7 seconds to open the project dialog box and pop up

**Comment by lalo-mx** [comment]
In my environment opening project dialog is also slow, using the profiler one method takes 24,319 ms
https://github.com/openjdk/jdk/blob/519ecd352a66633589f160db7390647d90e36b99/src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java#L897

![image](https://github.com/apache/netbeans/assets/7459881/22c7f26e-dc09-46d0-b346-2c9be9297442)


Product Version: Apache NetBeans IDE 20
Java: 21.0.1; OpenJDK 64-Bit Server VM 21.0.1+12-LTS
Runtime: OpenJDK Runtime Environment 21.0.1+12-LTS
System: Windows 10 version 10.0 running on amd64; UTF-8; en_US (nb)

**Comment by lyh061619** [comment]
My computer uses an MVME hard drive, which cannot be that slow, but using the Netbeans IDE is slow. When checking the monitor, most threads are waiting or timeout, and the experience is very poor

**Comment by mbien** [comment]
does this also happen with the open file dialog?

both file choosers open instantly for me with no delay.

I am wondering if this is a variant of https://github.com/apache/netbeans/discussions/6361, or some of the other JDK filechooser issues connected to disktop shortcuts on windows.

**Comment by lyh061619** [comment]
> does this also happen with the open file dialog?
> 
> both file choosers open instantly for me with no delay.
> 
> I am wondering if this is a variant of #6361, or some of the other JDK filechooser issues connected to disktop shortcuts on windows.
> 
> how long does this take to open?
> 
> ```java
>     public static void main(String[] args) {
>         JFileChooser jfs = new JFileChooser();
>         jfs.showOpenDialog(null);
>     }
> ```
Yes, I will use the test code you provided to sample the results as follows:
 
![m4](https://github.com/apache/netbeans/assets/15118571/2ba8eb25-dc8b-4579-9ccf-6d2890cc469d)

**Comment by mbien** [comment]
@lalo-mx `Win32ShellFolder` is used for the quick selection panel of the file chooser I believe.

there are multiple known issues connected to windows specific JDK code in that area, for example https://bugs.openjdk.org/browse/JDK-8305072
some are triggered by certain folders or links on your desktop.

Can you try to run NetBeans with: `-J-Dnb.FileChooser.useShellFolder=false`?

**Comment by lalo-mx** [comment]
> @lalo-mx `Win32ShellFolder` is used for the quick selection panel of the file chooser I believe.
> 
> there are multiple known issues connected to windows specific JDK code in that area, for example https://bugs.openjdk.org/browse/JDK-8305072 some are triggered by certain folders or links on your desktop.
> 
> Can you try to run NetBeans with: `-J-Dnb.FileChooser.useShellFolder=false`?

Yes, that fixes it in my environment

**Comment by binh190305** [comment]
-J-Dnb.FileChooser.useShellFolder=false i have tried this code by add it in "D:\NetBeans-21\netbeans\etc\netbeans.conf" in netbeans_default_options line and it has fixed the bug that make open file or add library so slow

**Comment by mbien** [comment]
thanks for confirming that turning off `ShellFolder` usage fixes the performance bug.

Upcoming NB 25 will use that API slightly less https://github.com/apache/netbeans/pull/8110 but those affected will likely still see delays before the directory chooser opens. So feel free running with the mentioned flag set, all it does is to remove the shortcuts panel as minor inconvenience.

It is also recommended to keep your JDK up to date since not all JDK bugs are backported to older major releases. (check the download page for the max supported JDK version for your installed NB release). (also take a look at the community installers, which bundle a fitting JDK with each release)
